src/FlatButton/FlatButton.js

```diff
@@ -6,7 +6,7 @@ import EnhancedButton from '../internal/EnhancedButton';
 import FlatButtonLabel from './FlatButtonLabel';
 
 function validateLabel(props, propName, componentName) {
-  if (!props.children && !props.label && !props.icon) {
+  if (!props.children && (!props.label && props.label !== 0) && !props.icon) {
```

Comment:
I'm wondering if we couldn't
- use the `warning` module to remove some of the code in production
- use `React. PropTypes.node` to improve this validation method.